operator/builtin/output/newrelic/payload.go

```diff
@@ -0,0 +1,93 @@
+package newrelic
+
+import (
+	"github.com/observiq/stanza/entry"
+	"github.com/observiq/stanza/internal/version"
+)
+
+// LogPayloadFromEntries creates a new []*LogPayload from an array of entries
+func LogPayloadFromEntries(entries []*entry.Entry) LogPayload {
+	logs := make([]*LogMessage, 0, len(entries))
+	for _, entry := range entries {
+		logs = append(logs, LogMessageFromEntry(entry))
+	}
+
+	lp := LogPayload{{
+		Common: LogPayloadCommon{
+			Attributes: map[string]interface{}{
+				"plugin": map[string]interface{}{
+					"type":    "stanza",
+					"version": version.GetVersion(),
+				},
+			},
+		},
+		Logs: logs,
+	}}
+
+	return lp
+}
+
+// LogPayload represents a single payload delivered to the New Relic Log API
+type LogPayload []struct {
+	Common LogPayloadCommon `json:"common"`
+	Logs   []*LogMessage    `json:"logs"`
+}
+
+// LogPayloadCommon represents the common attributes in a payload segment
+type LogPayloadCommon struct {
+	// Milliseconds or seconds since epoch
+	Timestamp int `json:"timestamp,omitempty"`
+
+	Attributes map[string]interface{} `json:"attributes"`
+}
+
+// LogMessageFromEntry creates a new LogMessage from a given entry.Entry
+func LogMessageFromEntry(entry *entry.Entry) *LogMessage {
+	logMessage := &LogMessage{
+		Timestamp:  entry.Timestamp.UnixNano() / 1000 / 1000, // Convert to millis
+		Attributes: make(map[string]interface{}),
+	}
+
+	// Promote message to the top-level
+	switch r := entry.Record.(type) {
+	case string:
+		logMessage.Message = r
+	case []byte:
+		logMessage.Message = string(r)
+	case map[string]interface{}:
+		// Instead of modifying the original with delete(), copy
+		// each key/value, promoting message if we come across it
+		for k, v := range r {
+			if k != "message" {
```

Comment:
Is there anywhere else in stanza that we are treating "message" as a special field in operator?

I get that we need to promote something _to_ message for the sake of New Relic, but it seems a bit odd that we would put this expectation on the the incoming format. 

I'd be curious to explore alternatives here. For instance, maybe we should consider adding a field parameter to the config. Similarly, an ordered (prioritized) list of fields where the first field that resolves to a string/[]byte is promoted?

Reply:
I took this behavior from the New Relic output plugin for FluentD. We do something similar to this in Nozzle (not in Stanza), but I agree that this should likely be configurable. And if not configurable, at least well-documented. What do you think about a config field that defaults to "message"?

Reply:
👍  That seems reasonable